subsys/testsuite/ztest/include/ztest_test_new.h

```diff
@@ -0,0 +1,323 @@
+	static STRUCT_SECTION_ITERABLE(ztest_unit_test, z_ztest_unit_test_##suite##_##fn) = {      \
+		.test_suite_name = STRINGIFY(suite),                                               \
+		.name = STRINGIFY(fn),                                                             \
+		.test = (_##suite##_##fn##_wrapper),                                               \
+		.thread_options = t_options,                                                       \
+	};                                                                                         \
+	static void _##suite##_##fn##_wrapper(void *data)                                          \
+	{                                                                                          \
+		ARG_UNUSED(data);                                                                  \
+		suite##_##fn();                                                                    \
+	}                                                                                          \
+	static inline void suite##_##fn(void)
+
+#define Z_ZTEST_F(suite, fn, t_options)                                                            \
+	static void _##suite##_##fn##_wrapper(void *data);                                         \
+	static void suite##_##fn(struct suite##_fixture *this);                                    \
+	static STRUCT_SECTION_ITERABLE(ztest_unit_test, z_ztest_unit_test_##suite##_##fn) = {      \
+		.test_suite_name = STRINGIFY(suite),                                               \
+		.name = STRINGIFY(fn),                                                             \
+		.test = (_##suite##_##fn##_wrapper),                                               \
+		.thread_options = t_options,                                                       \
+	};                                                                                         \
+	static void _##suite##_##fn##_wrapper(void *data)                                          \
+	{                                                                                          \
+		suite##_##fn((struct suite##_fixture *)data);                                      \
+	}                                                                                          \
+	static inline void suite##_##fn(struct suite##_fixture *this)
+
+#define ZTEST(suite, fn) Z_ZTEST(suite, fn, 0)
+
+#ifdef CONFIG_USERSPACE
+#define ZTEST_USER(suite, fn) Z_ZTEST(suite, fn, K_USER)
+#else
+#define ZTEST_USER(suite, fn)                                                                      \
+	BUILD_ASSERT(false, "Can't create userspace test without CONFIG_USERSPACE being enabled")
+#endif
+
+#define ZTEST_AUTOSPACE(suite, fn) Z_ZTEST(suite, fn, COND_VAR_1(CONFIG_USERSPACE, (K_USER), (0)))
+
+/**
+ * @brief Define a test function
+ *
+ * @param suite The test suite to attach this function to
+ * @param fn Test function
+ */
+#define ZTEST_F(suite, fn) Z_ZTEST_F(suite, fn, 0)
```

Comment:
Doxygen is missing information about fixture parameter and that it has fixed name `this`.

Reply:
Done